File: api/src/main/java/eu/suro/api/user/UserManager.java
package eu.suro.api.user;


import java.util.stream.Stream;

public interface UserManager<P> {

    /**
        * Получение всех пользователей
        * @return поток пользователей
     */
    default Stream<IUser<P>> getUsers() {
        throw new UnsupportedOperationException();
    }

    /**
        * Получение пользователя по игроку
        * @param player - игрок
        * @return пользователь
     */
    default IUser<P> getUser(P player) {
        throw new UnsupportedOperationException();
    }

}

File: api/src/main/kotlin/eu/suro/api/user/UserManager.kt
package eu.suro.api.user

import java.util.stream.Stream


class UserManager {
    /**
     * Получение всех пользователей
     * @return поток пользователей
     */
    fun getUsers(): Stream<IUser> {
        throw UnsupportedOperationException()
    }

    /**
     * Получение пользователя по игроку
     * @param player - игрок
     * @return пользователь
     */
    fun getUser(player: Any): IUser? {
        throw UnsupportedOperationException()
    }
}
